### DISCUSSION TITLE
Use up-target to replace multiple (matching) elements

### DISCUSSION THREAD

**Discussion by jwaschkau** [unpoly/unpoly#663] [Q&A]
Hi everyone,

I am in need of support implementing or configuring up-target to replace (update) multiple elements with the same up-id.
On my page there are multiple elements with the same content and the same up-id. Unpoly replaces only the one where the navigation action was started, as described in the documentation of fragment matching.

Is this possible? Should I not be using up-id but another identifier? And what would be the correct approach?

**Comment by adam12** [comment]
You can use `[up-id]` but similar to `[id]` attributes, they should be unique.

If you need to target multiple elements, you should build up a list of selectors that are comma-separated. `up-target="#one, #two, three"` etc.

**Reply by jwaschkau** [reply]
Hey Adam,

thanks a lot for taking the time answering questions.

Adding multiple selectors to up-target would not work in this case, as there can be 1 - n elements that should match and the backend does not know about the shown elements at the client side.

**Reply by adam12** [reply]
You could maybe add a listener for [`up:fragment:loaded`](https://unpoly.com/up:fragment:loaded), see if it's a fragment that needs to be replaced 1-n and then write your own implementation that calls [`up.render()`](https://unpoly.com/up.render).

This isn't typical usage of Unpoly that I've seen so you may run into other issues, but I don't see why it wouldn't work.

**Reply by jwaschkau** [reply]
Maybe it could be an improvement when multiple targets with the same selector can be updated. In the unpoly demo, if i change a company name from the project modal-dialog, the name is updated only in the child dialog and not in the list.

While updating across layers may be problematic, if there was a link to the company edit dialog from the list row in the all projects table, rows would possibly show the wrong company names, which could only be corrected by a complete table reload.

Maybe a new pseudo selector like :many could help?

I will try to extend this functionality with a custom up:fragment:loaded listener and report back.

**Reply by jwaschkau** [reply]
Adding the [up-hungry ](https://unpoly.com/up-hungry) attribute and setting the elements to the same [up-id](https://unpoly.com/up-id) seems to work fine. No custom code is needed here (Unpoly 3.10.2). The only problem could be that there are no guarantees it will work in newer versions?
